internal/service/systemd.go

```diff
@@ -55,12 +64,13 @@ type SystemdManager interface {
 }
 
 type systemdManager struct {
-	svcFilePath string
-	lock        sync.RWMutex
-	services    map[string]Service
+	svcFilePath   string
+	lock          sync.RWMutex
+	services      map[string]Service
+	eventListener *EventListener
 }
 
-func NewSystemdManager(configDir string) (SystemdManager, error) {
+func NewSystemdManager(configDir string, observerCh chan *Event) (SystemdManager, error) {
```

Comment:
who owns this channel? Why is coming from the upper layer when it is used internally?

Reply:
right now it's created by the workloadManager as part of the new instance logic:
https://github.com/project-flotta/flotta-device-worker/pull/217/files#diff-27f5b4db08ed174b7b18101c679d78c4dcf69db6fd8d6772f6a462f0fd6ae453R86

I'm going to refator this and pass the listener to the `NewSystemdManager` instead of hiding it inside that function. I think you will like it more :wink:

Reply:
Done. Please take a look.